src/tiled/abstracttool.cpp

```diff
@@ -81,21 +83,24 @@ void AbstractTool::setMapDocument(MapDocument *mapDocument)
         return;
 
     if (mMapDocument) {
-        disconnect(mMapDocument, &MapDocument::layerChanged,
-                   this, &AbstractTool::updateEnabledState);
-        disconnect(mMapDocument, &MapDocument::currentLayerChanged,
-                   this, &AbstractTool::updateEnabledState);
+        disconnect(
+            mMapDocument, &MapDocument::layerChanged, this, &AbstractTool::updateEnabledState);
+        disconnect(mMapDocument,
```

Comment:
it's what I mean under controversial, two very similar disconnects have absolutely different calls.